src/SwipeButton.tsx

```diff
@@ -63,140 +77,177 @@ export type SwipeButtonCommonProps = {
 
 const SwipeButton = ({
-    height = DEFAULT_HEIGHT, width = DEFAULT_WIDTH, borderRadius = DEFAULT_BORDER_RRADIUS,
-    title, titleContainerProps, titleProps, titleContainerStyle, titleStyle,
-    completeThresholdPercentage = DEFAULT_COMPLETE_THRESHOLD_PERCENTAGE,
-    underlayStyle, disabled, Icon, containerStyle,
-    onComplete, onSwipeEnd = () => {}, onSwipeStart = () => {},
+  height = DEFAULT_HEIGHT,
+  borderRadius = DEFAULT_BORDER_RADIUS,
+  title,
+  titleContainerProps,
+  titleProps,
+  titleContainerStyle,
+  titleStyle,
+  completeThresholdPercentage = DEFAULT_COMPLETE_THRESHOLD_PERCENTAGE,
+  underlayStyle,
+  disabled,
+  Icon,
+  iconContainerStyle,
+  containerStyle,
+  onComplete,
+  onSwipeEnd = () => {},
+  onSwipeStart = () => {},
 }: SwipeButtonProps) => {
-
-    const [endReached, setEndReached] = useState<boolean>(false);
-    const opacity = disabled ? 0.5 : 1;
-    const opacityStyle = { opacity };
-    const [translateX] = useState<Animated.Value>(new Animated.Value(0));
-    const scrollDistance = width - (completeThresholdPercentage / 100) - height;
-    const completeThreshold = scrollDistance * (completeThresholdPercentage / 100);
-
-    const animateToStart = () => {
-        Animated.spring(translateX, { toValue: 0, tension: 10, friction: 5, useNativeDriver: false }).start();
-
-        return setEndReached(false);
-    };
-
-    const animateToEnd = () => {
-        onComplete();
-        Animated.spring(translateX, { toValue: scrollDistance, tension: 10, friction: 5, useNativeDriver: false }).start();
-
-        return setEndReached(true);
-    };
-
-    const onMove = (_: GestureResponderEvent, gestureState: PanResponderGestureState) => {
-        if (disabled) {
-            return false;
-        }
-
-        if (gestureState.dx < 0 || gestureState.dx > scrollDistance) {
-            return Animated.event([{ dx: translateX }], { useNativeDriver: false })({
-                ...gestureState,
-                dx: gestureState.dx < 0 ? 0 : scrollDistance,
-            });
-        }
-        
-        return Animated.event([{ dx: translateX }], { useNativeDriver: false })(gestureState);
-    };
-
-    const onRelease = () => {
-        if (disabled) {
-            return;
-        }
-
-        if (endReached) {
-            return animateToStart();
-        }
-
-        const isCompleted = translateX._value! >= completeThreshold;
-        
-        return isCompleted
-            ? animateToEnd()
-            : animateToStart();
-    };
-
-    const panResponser = () => PanResponder.create({
-        onPanResponderGrant: onSwipeStart,
-        onPanResponderEnd: onSwipeEnd,
-        onStartShouldSetPanResponder: () => false,
-        onMoveShouldSetPanResponder: () => true,
-        onMoveShouldSetPanResponderCapture: () => false,
-        onPanResponderMove: onMove,
-        onPanResponderRelease: onRelease,
+  const [dimensions, setDimensions] = useState({ width: 0, height: 0 });
```

Comment:
To do some performance enhancements and eliminate re-renders, would it be possible to wrap some of these values in a `useMemo`?

Reply:
Good idea! I can take a look into adding this in.